client/App.jsx

```diff
@@ -1,23 +1,79 @@
 import React from 'react';
+import axios from 'axios';
 
 import ReviewsListContainer from './Rating&Reviews/ReviewsListContainer.jsx';
 import QuestionsAndAnswers from './Questions&Answers/QuestionsAndAnswers.jsx';
 
 class App extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      productList: [
+        {"id":17762,"campus":null,"name":null,"slogan":null,"description":null,"category":null,"default_price":null,"created_at":null,"updated_at":null}
+      ],
+      productId: 17762,
+      selectProductId: 17762,
+      selectProductInfo: [
+        {"id":17762,"campus":null,"name":null,"slogan":null,"description":null,"category":null,"default_price":null,"created_at":null,"updated_at":null}
+      ]
+    };
+    this.retrieveAllProductInfo = this.retrieveAllProductInfo.bind(this);
+    this.retrieveSelectProductInfo = this.retrieveSelectProductInfo.bind(this);
+  }
+
+  componentDidMount() {
+    this.retrieveSelectProductInfo();
+    this.retrieveAllProductInfo();
+  }
+
+  retrieveAllProductInfo() {
+    axios
+      .get(`/products`)
+      .then((response) => {
+        this.setState({
+          productList: response.data
+        })
+      })
+      .catch((error) => {
+        console.log('Get total product list failed...', error);
+      })
+  }
+
+  retrieveSelectProductInfo() {
+    const { selectProductId } = this.state;
+    axios
+      .get(`/products/${selectProductId}`)
+      .then((response) => {
+        this.setState({
+          selectProductInfo: response.data
+        })
+      })
+      .catch((error) => {
+        console.log('Get product data by id failed...', error);
+      })
   }
 
   render() {
+    const { selectProductId } = this.state;
+    const { selectProductInfo } = this.state;
+    const { productId } = this.state;
+    const { productList } = this.state;
```

Comment:
These lines can actually be combined into 

`{ selectProductId, selectProductInfo, productId, productList } = this.state`

It's a bit of a lengthy line, so changing it is just a stylistic choice, but I thought I'd let you know about the ability to destructure multiple items at once for future reference!